### ISSUE TITLE
How to describe dynamic keys in response schema

### ISSUE THREAD

**Issue by satkunas** [OAI/OpenAPI-Specification#1505]
How could the following response schema properties be described in YAML?

```
   "192.168.0.0/24": {
        "members": {
            "10:1f:74:b2:f6:a5": "192.168.0.2"
        }
}
```

Both the network address key `192.168.0.0/24` and each of the member key(s) `10:1f:74:b2:f6:a5` are dynamic and must follow a specific regular expression pattern. I've found documentation on how to define regular expression patterns for the value, but there's no details on how to define a pattern for the key.

**Comment by handrews** [comment]
@satkunas this is not currently possible in OpenAPI's Schema Object.

In normal JSON Schema you could use `patternProperties`, but OpenAPI forbids it.  If JSON Schema draft-06 or later were supported you could use `propertyNames`, but OpenAPI is based on draft-04

**Comment by handrews** [comment]
See also #1443 (alternate schema languages)

**Comment by tylermauthe** [comment]
Hello! This is a very clear and helpful explanation, thank you!

I was curious if there are any plans to move to JSON schema draft-06 or later? Cheers.

**Comment by levidavidmurray** [comment]
I was able to get this working by using `additionalProperties`.

```
components:
  schemas:
    MyObject:
      ...
    DynamicKeyObject:
      type: object,
      additionalProperties:
        type: array
        items:
          $ref: "#/components/schemas/MyObject"
```
and then my response is just
```
...
content:
  application/json:
    schema:
      $ref: "#/components/schemas/DynamicKeyObject"
```

Generating to typescript-axios results in 

```
export interface DynamicKeyObject {
  [key: string]: Array<MyObject>;
}
```

**Comment by handrews** [comment]
@tylermauthe @satkunas OpenAPI 3.1 is moving to the latest JSON Schema draft (#1977), so this will be possible with regular JSON Schema features in that version.  There's no way to do it in 3.0.x, and none can be added due to the patch release compatibility requirements. 

Since it's fixed in 3.1 and can't be fixed in earlier versions, I'm going to go ahead and close this.
